app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarController.java

```diff
@@ -190,19 +190,20 @@ public CalendarItem getCalendarItemByStartAndEndTime(DateTime start, DateTime en
         return calendarDao.getCalendarItemByStartAndEndTime(startString, endString);
     }
 
-    public void importCalendar(CalendarRowSet myCalendarList) {
-
+    public void importCalendar(Events myCalendarList) {
         // Cleanup cache before importing
         removeCache();
 
         // reading xml
-        List<CalendarRow> myCalendar = myCalendarList.getKalendarList();
-        for (CalendarRow row : myCalendar) {
-            // insert into database
-            try {
-                replaceIntoDb(row);
-            } catch (Exception e) {
-                Utils.log(e);
+        List<Event> events = myCalendarList.getEvents();
+        if (events != null) {
+            for (Event event : events) {
+                // insert into database
+                try {
+                    replaceIntoDb(event);
```

Comment:
Ahhhh. Database transactions in a loop :unamused: 
Well, I think that's out of scope four your PR, oh well

Reply:
More importantly, it’s not logic that I wrote. But now that you mention it, I can’t un-see it. Updated the DAO from `insert(CalendarItem)` to `insert(CalendarItem...)`.